README.md

```diff
@@ -1,3 +1,112 @@
-# Feature flag backend
+# Feature Flag Backend
 
-This is the repo for feature flag backend.
+The Feature Flag Backend service helps manage feature flags for your applications in Real Dev Squad. With feature flags, we can easily enable or disable specific features for different groups of users. It provides APIs for creating, updating, and retrieving feature flags and user mappings. The backend is built using Golang and is deployed using a serverless architecture on AWS Lambda. It uses DynamoDB as the database. Whenever we want to roll out new features gradually or experiment with different options, the Feature Flag Backend simplifies the process and gives us full control over our features.
+
+## Table of Contents
+
+-   [Installation](#installation)
+-   [Run](#run)
+-   [Usage](#usage)
+-   [Features](#features)
+-   [API Endpoints](#api-endpoints)
+-   [Data Model](#data-model)
+-   [Contributing](#contributing)
+
+## Installation
+You should have some things pre-installed :
+- [VS Code](https://code.visualstudio.com/) or any other IDE
+- [Git](https://git-scm.com/)
+- [Golang](https://go.dev/)(version 1.20 or later)
+- [Docker](https://www.docker.com/)
+
+
+1. **Clone the repository**
+
+    - **Open the terminal or command prompt:** Depending on your operating system, open the terminal or command prompt to begin the cloning process. 
+    - **Navigate to your desired local directory:** Use the cd command to navigate to the directory where you want to store the cloned repository.
+    - **Clone the repository:** Use the following command to clone the repository :
+        ```
+        git clone https://github.com/Real-Dev-Squad/feature-flag-backend.git
+        ```
+
+2. **AWS CLI**
+    - Follow all the steps mentioned in [AWS SAM prerequisites](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/prerequisites.html) document guide to install and setup AWS CLI
+    
+    > **Note**
+    > This step will not be required once the support for local DynamoDB setup is added. To know more read [this](https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/DynamoDBLocal.html)
+  
+3. **AWS SAM CLI**
+    - Follow all the steps mentioned for your local OS in [Installing the AWS SAM CLI](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/prerequisites.html) document guide to install and setup AWS SAM CLI
+
+## Run
```

Comment:
One imp step we should add is to configure their AWS profile and create database or if local db is successfully configured we can add that. As people will not be able to see the results without connecting to a database.

Reply:
> we should add is to configure their AWS profile

The steps to configure AWS profile are already covered in a step mentioned in the document linked under the install AWS CLI step

> create database
Updated the readme  with create database steps